include/eve/arch/riscv/as_register.hpp

```diff
@@ -0,0 +1,457 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/arch/riscv/predef.hpp>
+#include <eve/traits/as_integer.hpp>
+
+#include <type_traits>
+#if defined(EVE_HW_RISCV_SVE)
+namespace eve
+{
+template<typename T> struct logical;
+}
+
+namespace eve
+{
+template<typename Type, typename Size, rvv_abi ABI> struct as_register<Type, Size, ABI>
+{
+  static constexpr auto lmul = riscv_rvv_dyn_::getLMUL<Type>(Size::value);
+
+  static constexpr bool is_fp_v     = std::is_floating_point_v<Type>;
+  static constexpr bool is_signed_v = std::is_signed_v<Type>;
+
+  static consteval auto find_vint_mf8()
```

Comment:
there was some description somewhere, to what all this `m` mean - can you share please? I don't understand the docs well

Reply:
RISC-V vector instrinsic types that are correspond for integer/float types, constructed with:
1)  the part that shows the underlying element type: `vint` - signed integer, `vuint` - unsigned integer, `vfloat` - float or double, depending on SEW
2) the part that shows `EEW` (effective element width) of one element - it can be 8, 16, 32, 64, with some restrictions, e.g. you can not have float with SEW=8. Note: I may sometimes refer to `SEW` - single element width, in terms of intrinsics they behave the same or almost the same, the difference appears on asm level, as there are some  vector instructions, that "hardcode" SEW to another value.  
3) the part that shows `EMUL`(effective vector register group multiplier) configuration (or what part of real vector register should be used to form type). Again, I can say `LMUL`(vector register group multiplier), and in terms of intrinsics this is the same, the difference again appear on asm level, as when you "hardcode" `SEW`, `LMUL` is changed too. There we have 
3.1) `mf8` - this is fractional usage 1/8 of real vector register
3.2) `mf4` - this is fractional usage 1/4 of real vector register
3.3) `mf2` - this is fractional usage 1/2 of real vector register
3.4) `m1` - usage of 1 real vector register
3.5) `m2` - usage of 2 real vector registers
3.6) `m4` - usage of 4 real vector registers
3.7) `m8` - usage of 8 real vector registers

Note, that some combinations are unsupported. For example, there is no type `vint32mf8_t`, as this will require having this type for every vlen, including vlen=64. But for this vlen we have 1/8 of vector register equals to 8, that can not fit int.

For supported types you can check https://github.com/riscv-non-isa/rvv-intrinsic-doc/releases/tag/v1.0-rc2 `Chapter 7. Type system`. Note: part `7.4. Tuple type` is not interested in context of this review.

Reply:
Done